packages/types/src/utils.ts

```diff
@@ -96,3 +96,13 @@ export type Autocomplete<U extends T, T = string> = U | (T & Record<never, never
 export type Without<T, W> = {
   [P in keyof T as Exclude<P, W>]: T[P];
 };
+
+export type ResultWithError<R, E extends Error> =
+  | {
+      result: R;
```

Comment:
Let's rename this to data, error to align with BAPI client. The error is also a result in a way. Data is a generic positive name, but it's generic enough that it forces developers to rename it on their end. This is similar to what `useSWR` does.